protocol/testing/e2e/gov/vest_test.go

```diff
@@ -0,0 +1,254 @@
+					TreasuryAccount: "random_treasury",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 2, 0, 0, 0, 0, time.UTC),
+					EndTime:         time.Date(2024, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectSubmitProposalFail: true,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			tApp := testapp.NewTestAppBuilder().WithGenesisDocFn(func() (genesis types.GenesisDoc) {
+				genesis = testapp.DefaultGenesis()
+				testapp.UpdateGenesisDocWithAppStateForModule(
+					&genesis,
+					func(genesisState *govtypesv1.GenesisState) {
+						genesisState.Params.VotingPeriod = &testapp.TestVotingPeriod
+					},
+				)
+				// If updating an existing vest entry, initialize vest module with a vest entry that has the
+				// same key. Otherwise, initialize vest module with no vest entry.
+				genesisVestEntries := []vesttypes.VestEntry{}
+				if tc.updateExistingVestEntry {
+					genesisVestEntries = append(genesisVestEntries, vesttypes.VestEntry{
+						VesterAccount:   tc.msg.Entry.VesterAccount,
+						TreasuryAccount: tc.msg.Entry.TreasuryAccount + "_initial",
+						Denom:           tc.msg.Entry.Denom + "_initial",
+						StartTime:       tc.msg.Entry.StartTime.AddDate(0, 0, 1),
+						EndTime:         tc.msg.Entry.EndTime.AddDate(0, 0, 1),
+					})
+				}
+				testapp.UpdateGenesisDocWithAppStateForModule(
+					&genesis,
+					func(genesisState *vesttypes.GenesisState) {
+						genesisState.VestEntries = genesisVestEntries
+					},
+				)
+				return genesis
+			}).WithTesting(t).Build()
+			ctx := tApp.InitChain()
+			initialVestEntries := tApp.App.VestKeeper.GetAllVestEntries(ctx)
+
+			// Submit and tally governance proposal that includes `MsgSetVestEntry`.
+			ctx = testapp.SubmitAndTallyProposal(
+				t,
+				ctx,
+				&tApp,
+				[]sdk.Msg{tc.msg},
+				tc.expectCheckTxFails,
+				tc.expectSubmitProposalFail,
+				tc.expectedProposalStatus,
+			)
+
+			if tc.expectCheckTxFails || tc.expectSubmitProposalFail ||
+				tc.expectedProposalStatus == govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED {
+				// If governance proposal is not supposed to pass, verify that vest entries in state match
+				// vest entries before proposal submission.
+				require.Equal(t, initialVestEntries, tApp.App.VestKeeper.GetAllVestEntries(ctx))
+			} else {
+				// If proposal is supposed to pass, verify that expected vest entry is set in state.
+				vestEntry, err := tApp.App.VestKeeper.GetVestEntry(ctx, tc.msg.Entry.VesterAccount)
+				require.NoError(t, err)
+				require.Equal(t, tc.msg.Entry, vestEntry)
+			}
+		})
+	}
+}
+
+func TestDeleteVestEntry(t *testing.T) {
+	tests := map[string]struct {
+		msg                      *vesttypes.MsgDeleteVestEntry
+		vestEntryExists          bool // whether vest entry in above msg exists.
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success": {
+			msg: &vesttypes.MsgDeleteVestEntry{
```

Comment:
similar to above, 
1. define + use constants
2. write multiple vest entry deletion if possible